.github/workflows/ci.yaml

```diff
@@ -0,0 +1,47 @@
+name: CI
+
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+
+concurrency:
+  group: ${{ github.ref_name }}
+  cancel-in-progress: true
+
+jobs:
+    
+  rubocop:
+    name: Rubocop
+    uses: theforeman/actions/.github/workflows/rubocop.yml@v0
+
+  test:
+    name: Tests
+    needs: rubocop
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        ruby:
+          - "2.7"
+        puppet:
+          - "7"
+          - "8"
```

Comment:
I don't think Puppet 8 & Ruby 2.7 is a valid combination. For now I'd leave it out. I was only suggesting to make sure the rest of the CI workflow is ready for it.
```suggestion
```

Reply:
@ekohl oh i missed it, you were talking about matrix combination in this way? (i guess we can keep it, then!)
  ```
    matrix:
        ruby:
          - "2.7"
          - "3.1"
        puppet:
          - "7"
          - "8"
        include:
          - ruby: "2.7"
            puppet: "7"
          - ruby: "3.1"
            puppet: "8"
```

also i was looking for that puppet doc where i can see the compatible ruby version with the puppet version, can you help me with it?

Reply:
For this specific PR I would keep Puppet 8 and Ruby 3 out of consideration. Limit it to porting over what's already in Jenkins. There's a good chance you find more when you start adding Ruby 3 & Puppet 8 tests. Getting this in is a better first step. Otherwise you risk the PR dragging on for a long time.

Reply:
okay ack! thanks for the info.

Reply:
okay ack! thanks for the info.